bpf/http_sock.c

```diff
@@ -233,17 +260,26 @@ int BPF_KPROBE(kprobe_tcp_sendmsg, struct sock *sk, struct msghdr *msg, size_t s
 
     bpf_dbg_printk("=== kprobe tcp_sendmsg=%d sock=%llx size %d===", id, sk, size);
 
-    pid_connection_info_t info = {};
+    send_args_t s_args = {
+        .size = size
+    };
 
-    if (parse_sock_info(sk, &info.conn)) {
+    if (parse_sock_info(sk, &s_args.p_conn.conn)) {
         //dbg_print_http_connection_info(&info.conn); // commented out since GitHub CI doesn't like this call
-        sort_connection_info(&info.conn);
-        info.pid = pid_from_pid_tgid(id);
+        sort_connection_info(&s_args.p_conn.conn);
+        s_args.p_conn.pid = pid_from_pid_tgid(id);
 
         if (size > 0) {
             void *iovec_ptr = find_msghdr_buf(msg);
             if (iovec_ptr) {
-                handle_buf_with_connection(&info, iovec_ptr, size, 0, TCP_SEND);
+                u64 sock_p = (u64)sk;
+                bpf_map_update_elem(&active_send_args, &id, &s_args, BPF_ANY);
+                bpf_map_update_elem(&active_send_sock_args, &sock_p, &s_args, BPF_ANY);
+                handle_buf_with_connection(&s_args.p_conn, iovec_ptr, size, NO_SSL, TCP_SEND);
+                // if (size < KPROBES_LARGE_RESPONSE_LEN) {
```

Comment:
This is some backup code I'd like to keep around if further tests find issues. We'll clean it up later.